src/coreclr/src/dlls/mscoree/coreclr/CMakeLists.txt

```diff
@@ -84,6 +84,23 @@ if (CLR_CMAKE_HOST_UNIX)
     set(LIB_UNWINDER unwinder_wks)
 endif (CLR_CMAKE_HOST_UNIX)
 
+if(CLR_CMAKE_HOST_WIN32)
+    set(COMPRESSION_LIB clrcompression-static)
```

Comment:
Nit: If we are changing clrcompression to be linked statically, we can also rename it to `System.IO.Compression.Native` to be symmetric with the rest.

Reply:
I did not mean to make clrcompression always linked. Sorry for the confusion. This is for testing.

These changes are relatively hard to hit in tests/debugger if only linked by the static host so I temporarily made them linked in coreclr as well. 
Hosting tests mostly stress the hosting/bundling aspect, but the code that they run is HeloWorld-level. I had situations before where there were obvious flaws in overriding and yet all tests were passing. 
When coreclr links something, normal CI tests will hit the overrides much more reliably.

Reply:
I will remove linking of `clrcompression` into coreclr.so     
I would not have objections if we want that linked though. We can easily do that now.